compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/CheckExpressionTest.java

```diff
@@ -36,15 +36,25 @@ public void testCheckExprPrecedence() {
         testFile("check-expr/check_expr_source_02.bal", "check-expr/check_expr_assert_02.json");
     }
 
-    // Recovery test
+    @Test
+    public void testSimpleCheckOnFail() {
+        testFile("check-on-fail/check_on_fail_source_01.bal",  "check-on-fail/check_on_fail_assert_01.json");
+    }
 
     @Test
     public void testCheckWithMissingExpr() {
         testFile("check-expr/check_expr_source_03.bal", "check-expr/check_expr_assert_03.json");
     }
 
+    // Recovery test
```

Comment:
Need to move this comment before the `testCheckWithMissingExpr()` as it is a recovery test